students/mxwllndrsn/lesson04/mailroom_mk2.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+
+#uw python 210
+#lesson 03
+#max anderson
+
+#mailroom.py
+
+import os
+import io
+from datetime import date
+
+donor_list = [
+              ['Andrew Jackson', 10000, 5, 2000],
+              ['Benjamin Franklin', 253000, 2, 117500],
+              ['John Kennedy', 20, 1, 20],
+              ['Abraham Lincoln', 807, 2, 403.5],
+              ['Jim Jones', 918, 1, 918]
+             ]
+
+# Index 0 - Donor Name
+# Index 1 - Donation Totals
+# Index 2 - Number of Donations
+# Index 3 - Average Donation
+
+ymdate = str(date.today())
+
+#implementing dictionary function menu
+def menu_select(prompt, menu):
+    while(True):
+        select = input(prompt)
+        while(select not in menu.keys()):
+            select = input('Please try again: ')
+        if menu[select]() == 'exit':
+            break
+
+
+def exit_menu():
+    print('Exiting...')
+    return 'exit'
+
+
+def manage_donors():
+    menu_select(manage_prompt, manage_menu)
+
+
+def enter_donation():
+    print()
+    name = input('Enter full name of \'Donor\': ').title()
+    if in_list(name): donation_update(name)
+    elif not in_list(name): donation_update(new_donor(name))
+
+    letter = input('Create thank you letter? (y/n) ').title()
+    if letter == 'Y': print(format_thx(name))
+    elif letter == 'N': return
```

Comment:
although it is valid, it is discouraged and will be flagged by a linter when using inline if/else statements.